File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: str = Field(max_length=20)
    content: str = Field(max_length=2000)


class ChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=1000)
    history: list[ChatMessage] = Field(default_factory=list, max_length=12)
    model: str | None = None


class ImageRequest(BaseModel):
    prompt: str = Field(min_length=3, max_length=500)
    model: str = "nano-banana"
    width: int = Field(default=1024, ge=256, le=1536)
    height: int = Field(default=1024, ge=256, le=1536)


class ChatResponse(BaseModel):
    reply: str
    remaining: int
    daily_limit: int
    limit_reached: bool = False
    payment_info: dict | None = None
    model: str | None = None


class ImageResponse(BaseModel):
    url: str | None = None
    model: str
    prompt: str
    provider: str | None = None
    error: str | None = None


class UsageResponse(BaseModel):
    used: int
    remaining: int
    daily_limit: int
    payment_info: dict


class PaymentInfoResponse(BaseModel):
    card_number: str
    card_holder: str
    bank_name: str
    description: str
    amount_suggested: int
    contact_email: str
    message: str


class ModelsResponse(BaseModel):
    chat_models: list[dict]
    image_models: list[dict]
    capabilities: list[str]
    status: dict | None = None


class AIStatusResponse(BaseModel):
    ready: bool
    chat_provider: str
    image_provider: str
    pollinations_configured: bool
    openrouter_configured: bool = False
    openai_configured: bool = False
    perplexity_configured: bool = False
    gemini_configured: bool
    providers: list[str] = []
    message: str
    limits: dict | None = None

File: frontend/src/lib/api/types/ai.ts
export interface PaymentInfo {
  card_number: string;
  card_holder?: string;
  bank_name?: string;
  amount_suggested: number;
  message: string;
}

export interface ChatMessage {
  role: "user" | "assistant" | "system";
  content: string;
}

export interface ChatResponse {
  reply: string;
  remaining: number;
  daily_limit: number;
  limit_reached: boolean;
  model?: string;
}

export interface AIModelOption {
  id: string;
  label: string;
  provider: string;
  desc?: string;
  fast?: boolean;
  smart?: boolean;
}

export interface AIStatus {
  ready: boolean;
  chat_provider: string;
  image_provider: string;
  pollinations_configured: boolean;
  openrouter_configured?: boolean;
  gemini_configured: boolean;
  providers?: string[];
  message: string;
  limits?: {
    site_daily: number;
    site_note: string;
    providers_note: string;
  };
}

export interface AIModelsInfo {
  chat_models: AIModelOption[];
  image_models: AIModelOption[];
  capabilities: string[];
  status?: AIStatus;
}

export interface ImageGenResult {
  url: string | null;
  model: string;
  prompt: string;
  provider?: string;
  error?: string;
}

export interface UsageInfo {
  used: number;
  remaining: number;
  daily_limit: number;
  payment_info?: PaymentInfo;
}
